### PR TITLE
Merge master into release/v3

### PR DESCRIPTION
**Pull request by fvoska**
This PR updates the release/v3 branch with changes from master.

TODO:

- [ ] Check why the tests are failing
- [x] `network` in `getDefaultConfig`?

### CODE REVIEW

File: packages/core/src/View.ts [hunk 1]
```diff
@@ -40,12 +39,14 @@ export class View<T extends PureModel = PureModel> extends ToMany<T> {
     );
     this.modelType = getModelType(modelType);
     this.sortMethod = sortMethod;
-    mobx.makeObservable(this, {
-      sortMethod: mobx.observable,
```
Comment (fvoska) [inline packages/core/src/View.ts:44]: TODO: is it ok that these options were removed?

File: packages/core/src/View.ts [hunk 2]
```diff
@@ -40,12 +39,14 @@ export class View<T extends PureModel = PureModel> extends ToMany<T> {
...
-      list: mobx.computed,
-    });
+    mobx.makeObservable(this);
+  }
+
+  public get length(): number {
+    return this.value.length;
   }
 
+  @mobx.computed
```
Comment (fvoska) [inline packages/core/src/View.ts:49]: TODO: Does this have to be computed? It wasn't before, was that incorrect?

File: packages/core/src/helpers/model/utils.ts [hunk 3]
```diff
@@ -185,13 +218,18 @@ export function modelToJSON(model: PureModel): IRawModel {
...
+      const modelClass = model.constructor as typeof PureModel;
+      const mapField = getMeta(modelClass, `${MetaClassField.MapField}_${fieldName}`, null, true);
+      const key = mapField ? mapField : fieldName;
+      // Make sure to use an existing value if the attribute is used multiple times
+      raw[key] = raw[key] || modelMapSerialize(modelClass, model, fieldName);
     }
   });
 
-  return mobx.toJS(raw);
+  // return mobx.toJS(raw);
```
Comment (fvoska) [inline packages/core/src/helpers/model/utils.ts:231]: I guess we can remove this line?

File: packages/network-core/src/Response.ts [hunk 4]
```diff
@@ -154,7 +154,7 @@ export class Response<T extends TModel | Array<TModel>, TModel extends PureModel
       try {
         this.__data = initData(response, collection, overrideData);
       } catch (e) {
-        this.__internal.error = e;
+        this.__internal.error = e as any;
```
Comment (fvoska) [inline packages/network-core/src/Response.ts:157]: Check if we can remove `as any` (I think there is another place with this identical line, probably in obsolete/network)

File: packages/network-core/test/basic.test.ts [hunk 5]
```diff
@@ -3,6 +3,8 @@ import * as nock from 'nock';
 import fetch from 'node-fetch';
 import { Network, NetworkClient, Response, setUrl } from '../src';
 
+console.log(nock('aaa'));
```
Comment (fvoska) [inline packages/network-core/test/basic.test.ts:6]: TODO: check why tests throw an error `TypeError: nock is not a function`. It causes this test suit to fail

File: packages/network-jsonapi/tsconfig.build.json [hunk 6]
```diff
@@ -14,11 +14,11 @@
     "noImplicitAny": false,
     "noImplicitReturns": true,
     "noUnusedParameters": true,
-    "outDir": "./dist",
+    "outDir": ".",
     "strict": true,
     "strictFunctionTypes": false,
-    "target": "es5",
-    "resolveJsonModule": true
+    "target": "es6",
```
Comment (fvoska) [inline packages/network-jsonapi/tsconfig.build.json:20]: TODO: most other configs use es5 target. Should we maybe bump all of them to es6?

File: packages/obsolete-datx-jsonapi-angular/package.json [hunk 7]
```diff
@@ -23,22 +23,23 @@
...
+    "jest": "^27.0.5",
     "lodash": "^4.17.20",
     "rollup": "^2.38.5",
     "rollup-plugin-exclude-dependencies-from-bundle": "^1.1.17",
     "rollup-plugin-terser": "^7.0.2",
-    "rollup-plugin-typescript2": "^0.29.0",
-    "rxjs": "^6.6.3",
-    "ts-jest": "^26.5.1",
-    "typescript": "^4.1.5",
+    "rxjs": "^7.0.0",
```
Comment (fvoska) [inline packages/obsolete-datx-jsonapi-angular/package.json:39]: TODO: check the whole project how we define peer and dev deps for RxJS. Angular >=12.2 works with RxJS 7 but older versions might not

File: packages/obsolete-datx-jsonapi-angular/src/Response.ts [hunk 8]
```diff
@@ -30,10 +31,18 @@ export class Response<T extends IJsonapiModel = IJsonapiModel> extends PromiseRe
...
       }
     }
 
     return this.__cache[name];
   }
+
+  public get snapshot(): IResponseSnapshot {
+    const snapshot = super.snapshot;
+
+    delete snapshot.options?.fetchOptions?.['takeUntil$'];
```
Comment (fvoska) [inline packages/obsolete-datx-jsonapi-angular/src/Response.ts:44]: TODO: with new networking, check if we can remove takeUntil$. I'd say we can

File: packages/utils/src/mobx.ts [hunk 9]
```diff
@@ -79,9 +83,10 @@ class MobXProxy {
   private hasMobX = false;
 
   constructor() {
+    let mobx: any;
```
Comment (fvoska) [inline packages/utils/src/mobx.ts:86]: `typeof require('mobx')` 😂

File: packages/core/src/View.ts (hunk 2 above)
Comment (fvoska) [reply packages/core/src/View.ts:49]: It does

**Review by DarkoKukovec — APPROVED** [review]

**Pull request merged**